extensions/wikia/Wall/WallExternalController.class.php

```diff
@@ -277,6 +277,10 @@ public function deleteMessage() {
 			break;
 
 			case 'remove':
+				if (!$mw->canModerate($this->wg->User)) {
```

Comment:
@garthwebb - Because `$mw` is made using `WallMessage::newFromId`, some of its internal attributes are never loaded - including the instance variable `$articleComment`. `WallMessage::canRestore` and `WallMessage::canRemove` both call `WallMessage::isAuthor` (see L911). If `$articleComment` is only created via `ArticleComment::newFromTitle`, none of _it's_ internal data is loaded either - resulting in the `$mUser` instance variable for the article comment to be null, meaning `ArticleComment::isAuthor` (https://github.com/WikiaVolDev/app/blob/master/extensions/wikia/ArticleComments/classes/ArticleComment.class.php#L581) will always return false.

To get around this, you need to use `WallMessage::load`, which will load all of the necessary data (including the base article comment for the wall message, and therefore the author of the message).

There's no reason to load the data if the current user _can_ moderate as the check for the `wallremove` permissions happen before `isAuthor` is called, and all the needed data for that check is passed as the function argument